### DISCUSSION TITLE
renaming files with dropzone

### DISCUSSION THREAD

**Discussion by prescriptionlifeline** [Laravel-Backpack/community-forum#1037] [Q&A (Help)] [answered]
Quoting https://backpackforlaravel.com/docs/6.x/crud-fields#dropzone-pro :

```
    // optional configuration.
    // check available options in https://docs.dropzone.dev/configuration/basics/configuration-options
    // 'configuration' => [
    //     'parallelUploads' => 2,
    // ]
```
So in https://docs.dropzone.dev/configuration/basics/configuration-options there's a `renameFile` option. https://stackoverflow.com/a/50056049/569976 shows how to use it but that example is in javascript whereas here I'm trying to do it through PHP.

I thought maybe this would work but it didn't:

```
            'configuration' => [
                'renameFile' => '() => crypto.randomUUID()',
            ],
```
For that matter I'd like to be able to preserve whatever extension the uploaded file had but it's not clear to me how to do that either altho I suppose _that_ might be a question better suited for the dropzone devs.

Any ideas?

**Comment by pxpm** [comment, marked as answer]
Hey @prescriptionlifeline 

If you are using Backpack uploaders, we do the file naming in PHP, so whatever you set in javascript will be overwritten when the file is uploaded.

You can configure the file name with: https://backpackforlaravel.com/docs/6.x/crud-uploaders#naming-files-when-using-uploaders-1

Let me know if that helps. 

Cheers

**Reply by prescriptionlifeline** [reply]
That worked! Here's what I wound up doing:

```php
            'withFiles' => ['fileNamer' => function($file, $uploader) { return generateRandomString(32) . '.' . $file->extension(); }],
```
I'll test out your other answer as time permits - gotta get ready for work.
